### PR TITLE
Fixes a bunch of warnings

### PR DESCRIPTION
**Pull request by csilvestrini**

### CODE REVIEW

File: src/runtime/testing/slot-test-observer.ts [hunk 1]
```diff
@@ -150,7 +150,9 @@ export class SlotTestObserver extends AbstractSlotObserver {
     if (this.onlyOptionalExpectations()) {
       return Promise.resolve();
     }
-    return new Promise((resolve, reject) => this.onExpectationsComplete = resolve);
+    return new Promise((resolve, reject) => {
+      this.onExpectationsComplete = resolve;
+    });
```
Comment (Cypher1) [inline src/runtime/testing/slot-test-observer.ts]: This seems like a case of the linter being overly restrictive, I'm not blocking on it, but maybe we should relax the lint rule?

File: src/tools/schema2kotlin.ts [hunk 2]
```diff
@@ -107,7 +107,7 @@ class KotlinGenerator implements ClassGenerator {
   fieldsReset: string[] = [];
   getUnsetFields: string[] = [];
 
-  constructor(readonly node: SchemaNode, private opts: minimist.ParsedArgs) {}
+  constructor(readonly node: SchemaNode, private readonly opts: minimist.ParsedArgs) {}
```
Comment (Cypher1) [inline src/tools/schema2kotlin.ts]: What's the linter's justification for this? Is it just that we don't write to it so we should be restrictive by default?

Thanks @csilvestrini

Comment (csilvestrini) [reply src/tools/schema2kotlin.ts]: That's right

File: src/runtime/testing/slot-test-observer.ts (hunk 1 above)
Comment (csilvestrini) [reply src/runtime/testing/slot-test-observer.ts]: I'm not sure that is possible.

**Comment by csilvestrini** [comment]
> Thanks Cameron, I appreciate quietening the linter, but most of these (not the s/level/lvl/ and Unit changes) seem like overly restrictive lint rules.

I'm not sure how much control we have over these rules. They're generally google defaults, and to some degree enforce google style guidelines. We could start trying to play whack-a-mole and disable the lint warnings as they turn up on a case-by-case basis if we happy disagree with them, but I think it's just easier to just keep our code compliant.

Removing the explicit Unit was an IntelliJ suggestion, which I happen to agree with. The rest were compiler warnings or errors from google lint tools.

**Comment by Cypher1** [comment]
100% on board with matching Google style. Thanks for checking :)

Joshua Pratt | Arcs | jopra@google.com

**Comment by csilvestrini** [comment]
Passing cl/290537142

**Pull request merged**